app/services/foreman/renderer/scope/macros/base.rb

```diff
@@ -416,14 +416,17 @@ def foreman_server_ca_cert(server_ca_file_enabled: true, ssl_ca_file_enabled: tr
             example 'join_with_line_break(values) # => 1\n2\n3 for CSV,JSON,YAML'
           end
           def join_with_line_break(array)
+            # Normalize elements of the array to end with "\n" if they do
+            # not do already.
+            array.map! { |str| str.end_with?("\n") ? str : "#{str}\n" }
```

Comment:
The unfortunate impact of this is that if you pass in a single-element array without a trailing newline, it will get added implicitly. Is that really what's expected here?

Reply:
Well, yes and no. I think the more generic question would be if we want the last line to have a line feed at the end. For HTML output I guess this depends on the context of the output. For tables we might not want this, for other outputs maybe. For text based output this may end up in philosophical discussions as I remember that some text editors typically add a line feed to the last line while others don't. I could live with it either way. Next question would be if we would remove a line feed at the end of the last element of the array if already present or if we should just leave the last element of the array as is.